### PR TITLE
Issue #2476 Add the host interface's IP to the noProxy list

### PR DESCRIPTION
**Pull request by svenstaro**

### CODE REVIEW

**Comment by svenstaro** [comment]
Please don't merge this right away. I would like to verify with a colleague beforehand.

**Comment by svenstaro** [comment]
@gbraad You're right. Added an explanatory comment.

**Comment by gbraad** [comment]
Added WIP state to prevent merging. Please verify the artifact after build passed ...

**Comment by gbraad** [comment]
add to whitelist

**Comment by svenstaro** [comment]
retest this please

**Comment by LalatenduMohanty** [comment]
@svenstaro PR does not work for me. I can only see the VM's IP address

```
$ minishift docker-env --no-proxy
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.64.3:2376"
export DOCKER_CERT_PATH="/Users/lmohanty/.minishift/certs"
export NO_PROXY="192.168.64.3"
# Run this command to configure your shell:
# eval $(minishift docker-env --no-proxy)
```

**Comment by praveenkumar** [comment]
@svenstaro you want to add host interface IP to the VM shell variable instead of host I think, right now what you are doing is adding it to host itself.

**Comment by gbraad** [comment]
@praveenkumar can you make the comment in code and provide the alternative/correct approach?

**Comment by gbraad** [comment]
@svenstaro @praveenkumar It depends on what he needs to achieve.

Either the host or the instance would need this no_proxy?

**Comment by svenstaro** [comment]
I want to make sure that even when a proxy is in use that services on the host can be reached directly by containers run inside minishift without trying to go through the proxy.

For instance, if a database ran on the host, I want to be able to reach that directly from inside a container.

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -354,7 +355,13 @@ func createHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
 			fmt.Println("FAIL")
 			return nil, fmt.Errorf("Error getting VM IP: %s", err)
 		}
+		hostIP, err := minishiftNetwork.DetermineHostIP(h.Driver)
```
Comment (praveenkumar) [inline pkg/minikube/cluster/cluster.go]: :+1:

File: pkg/minikube/cluster/cluster.go [hunk 2]
```diff
@@ -354,7 +355,13 @@ func createHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
...
+		if err != nil {
+			fmt.Println("FAIL")
+			return nil, fmt.Errorf("Error getting host IP: %s", err)
+		}
 		config.ShellProxyEnv.AddNoProxy(vmIP)
+		config.ShellProxyEnv.AddNoProxy(hostIP)
```
Comment (praveenkumar) [inline pkg/minikube/cluster/cluster.go]: :+1:

**Comment by svenstaro** [comment]
This is also verified working on Windows.

**Pull request merged**